src/Symfony/Component/Validator/ValidatorBuilder.php

```diff
@@ -213,8 +228,8 @@ public function disableAnnotationMapping()
      */
     public function setMetadataFactory(MetadataFactoryInterface $metadataFactory)
     {
-        if (count($this->xmlMappings) > 0 || count($this->yamlMappings) > 0 || count($this->methodMappings) > 0 || null !== $this->annotationReader) {
-            throw new ValidatorException('You cannot set a custom metadata factory after adding custom mappings. You should do either of both.');
+        if ($this->xmlMappings || $this->yamlMappings || $this->methodMappings || $this->extraLoaders || null !== $this->annotationReader) {
+            throw new ValidatorException('You cannot set a custom metadata factory after adding custom mappings or extra loaders. You should do either of both.');
```

Comment:
Do you have an alt suggestion?

Reply:
`You cannot use a custom metadata factory and custom mappings/loaders at the same time.` All of the current messages are misleading, because they imply order matters.